File: 20221130/JadenCase/src/test/java/JadenCaseTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class JadenCaseTest {
    @Test
    void solution() {
        JadenCase test = new JadenCase();

        assertEquals("3people Unfollowed Me", test.solution("3people unFollowed me"));
        assertEquals("For The Last Week", test.solution("for the last week"));
        assertEquals("For The  Last Week", test.solution("for the  last week"));
        assertEquals(" For The Last Week ", test.solution(" for the last week "));
    }

    @Test
    void changeJadenCase() {
        JadenCase test = new JadenCase();

        assertEquals("3people", test.changeJadenCase("3people"));
        assertEquals("3peo3le", test.changeJadenCase("3peo3le"));
        assertEquals("Unfollowed", test.changeJadenCase("unFollowed"));
        assertEquals("Me", test.changeJadenCase("me"));
    }
}

File: 2022/20221201/JadenCase.test.js
import solution, { changeJadenCase } from './JadenCase';

test('solution', () => {
  expect(solution('3people unFollowed me')).toBe('3people Unfollowed Me');
  expect(solution('for the last week')).toBe('For The Last Week');
  expect(solution('for the  last week')).toBe('For The  Last Week');
  expect(solution(' for the last week ')).toBe(' For The Last Week ');
});

test('changeJadenCase', () => {
  expect(changeJadenCase('3people')).toBe('3people');
  expect(changeJadenCase('unFollowed')).toBe('Unfollowed');
  expect(changeJadenCase('me')).toBe('Me');
  expect(changeJadenCase(' ')).toBe(' ');
  expect(changeJadenCase('  ')).toBe('  ');
});
